element-connector/src/main/java/org/eclipse/californium/elements/DtlsEndpointContext.java

```diff
@@ -47,7 +47,32 @@
 	 * The name of the attribute that contains the timestamp of the last
 	 * handshake of the DTLS session.
 	 */
-	public static final String KEY_HANDSHAKE_TIMESTAMP = "KEY_HANDSHAKE_TIMESTAMP";
+	public static final String KEY_HANDSHAKE_TIMESTAMP = "DTLS_HANDSHAKE_TIMESTAMP";
+	/**
+	 * The name of the attribute that contains a handshake mode. Values see
+	 * {@link #HANDSHAKE_MODE_FORCE_FULL}, {@link #HANDSHAKE_MODE_FORCE},
+	 * {@link #HANDSHAKE_MODE_START}, and {@link #HANDSHAKE_MODE_NONE}. None
+	 * critical attribute, not considered for matching.
+	 */
+	public static final String KEY_HANDSHAKE_MODE = KEY_PREFIX_NONE_CRITICAL + "DTLS_HANDSHAKE_MODE";
+
+	/**
+	 * Force full handshake before send this message.
+	 */
+	public static final String HANDSHAKE_MODE_FORCE_FULL = "full";
+	/**
+	 * Force handshake before send this message.
```

Comment:
should we add javadoc to explain that abbreviated handshake will be tried if a session is available ?